pkg/api/common/helper.go

```diff
@@ -351,3 +353,69 @@ func WrapAsParameter(s string) string {
 func WrapAsVerbatim(s string) string {
 	return fmt.Sprintf("',%s,'", s)
 }
+
+// GetDockerConfig transforms the default docker config with overrides. Overrides may be nil.
+func GetDockerConfig(opts map[string]string, overrides []func(*DockerConfig) error) (string, error) {
+	config := GetDefaultDockerConfig()
+
+	for i := range overrides {
+		if err := overrides[i](&config); err != nil {
+			return "", err
+		}
+	}
+
+	dataDir, ok := opts[ContainerDataDirKey]
+	if ok {
+		config.DataRoot = dataDir
+	}
+
+	b, err := json.MarshalIndent(config, "", "    ")
+	return string(b), err
+}
+
+// GetContainerdConfig transforms the default containerd config with overrides. Overrides may be nil.
+func GetContainerdConfig(opts map[string]string, overrides []func(*ContainerdConfig) error) (string, error) {
```

Comment:
I came up with the functional setters to work around the fact that there are templated values in the docker/containerd config normally, but we don't want that logic in the helper functions (for testing purposes and circular imports).

I realize now that having both map[string]string and []func(*ContainerdConfig) error) is a bit redundant, for example we could have a separate helper with signature `func help(map[string]string) []func(*ContainerdConfig) error)` for containerd/docker each and use only functional setters to construct the final object